docs/book/v2/routing-middleware.md

```diff
@@ -0,0 +1,265 @@
+                'type' => Literal::class,
+                'options' => [
+                    'route' => '/container-handler-class-literal',
+                    'defaults' => [
+                        'controller' => PipeSpec::class,
+                        'middleware' => AlbumDetailHandler::class,
+                    ],
+                ],
+            ],
+            'container-handler' => [
+                'type' => Literal::class,
+                'options' => [
+                    'route' => '/container-handler',
+                    'defaults' => [
+                        'controller' => PipeSpec::class,
+                        'middleware' => 'container_id_for_album_detail_handler',
+                    ],
+                ],
+            ],
+            'container-pipe' => [
+                'type' => Literal::class,
+                'options' => [
+                    'route' => '/container-pipe',
+                    'defaults' => [
+                        'controller' => PipeSpec::class,
+                        'middleware' => AlbumDetailMiddlewarePipe::class,
+                    ],
+                ],
+            ],
+             'container-middleware' => [
+                 'type' => Literal::class,
+                 'options' => [
+                     'route' => '/container-middleware',
+                     'defaults' => [
+                         'controller' => PipeSpec::class,
+                         'middleware' => SomeMiddleware::class,
+                     ],
+                 ],
+             ],
+       ],
+    ],
+];
+```
+
+##  Anonymous Interface Implementations
+
+You can also pass an anonymous class which implements `MiddlewareInterface` or `RequestHandlerInterface` directly. This
+is primarily for development because it can cause issues with configuration caching.
+
+```php
+use Laminas\Diactoros\Response\TextResponse;
+use Laminas\Mvc\Middleware\PipeSpec;
+use Laminas\Router\Http\Literal;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+return [
+    'router' => [
+        'routes' => [
+            'instance-middleware' => [
+                'type' => Literal::class,
+                'options' => [
+                    'route' => '/instance-middleware',
+                    'defaults' => [
+                        'controller' => PipeSpec::class,
+                        'middleware' => new class implements MiddlewareInterface {
+                            public function process(
+                                ServerRequestInterface $request,
+                                RequestHandlerInterface $handler
+                            ): ResponseInterface {
+                                // You have to return a response here because attempting to use $handler here will
+                                // always result in an empty pipeline exception.
```

Comment:
```suggestion
                                // You have to return a response here because attempting to use $handler here will
                                // always result in an empty pipeline exception.
                                // This middleware pipe does not delegate to controllers or middleware from other routes. 
```